src/requestVariableCacheValueProcessor.ts

```diff
@@ -1,114 +1,83 @@
 "use strict";
-import { window } from "vscode";
-
-import { MimeUtility } from "./mimeUtility";
 import { RequestVariableCacheValue } from './models/requestVariableCacheValue';
 import { HttpResponse } from './models/httpResponse';
+import { HttpRequest } from "./models/httpRequest";
+
+const jp = require('jsonpath');
 
-const validPathRegex = new RegExp(/(\w+)(\.\w+|\[\d+\])*/);
-const partRegex = new RegExp(/(\w+)(\[\d+\])*/);
-const arrayIndexRegex = new RegExp(/\[(\d+)\]/);
+const requestVariablePathRegex: RegExp = /(\w+)(?:\.)?(response|request)?(?:\.)?(body|headers)?(?:\.)?(.*)?/;
```

Comment:
why add some `?` after some groups, they can be omitted?

Reply:
It's more about not needing a complete path in order to resolve a value. `var.response` is a valid path, `var.response.body` also. In other words, you don't need the complete `var.response.body.$.id` to resolve a value.

I'm not sure what use-cases it is, but it doesn't hurt to support it.

Reply:
In this way you can hover over `{{var.response.body}}` to see what `body` contains.

Reply:
@cbrevik in this case, we allow `body` on when `response` or `request` is provided, we can't write statements like `variable.body.$.id`, it should be `variable.response.body.$.id`.
So the regex maybe should be like this:
```js
/(\w+)(?:\.(request|response)(?:\.(body|headers)(?:\.(.*))?)?)?/
```

Reply:
Ah sure, I'll look into that.

Reply:
Updated